File: 2188.Minimized Maximum of Products Distributed to Any Store/solution.c
int candist(int n, int* quantities, int quantitiesSize, int prod) {
    int required = 0;
    for(int i=0;i<quantitiesSize;i++){
        required+=(quantities[i]+prod-1)/prod;
    }
    if(required>n) return false;
    return required<=n;
}

int minimizedMaximum(int n, int* quantities, int quantitiesSize) {
    int left = 1;
    int right = 0;
    for(int i=0;i<quantitiesSize;i++){
        if(right<quantities[i]){
            right = quantities[i];
        }
    }    
    int res = 0;

    while(left<=right){
        int mid = left +(right-left)/2;

        if(candist(n,quantities,quantitiesSize,mid)){
            res = mid;
            right = mid - 1;
        }else{
            left = mid + 1;
        }
    }
    return res;
}

File: 2188.Minimized Maximum of Products Distributed to Any Store/solution.cs
public class Solution {
    public bool checker(int n, int[] quantities, int quan){
        int store = 0;
        foreach(int i in quantities){
            store+=(i+quan-1)/quan;
        }
        return store<=n;

    }
    public int MinimizedMaximum(int n, int[] quantities) {
        int left = 1;
        int right = 0;
        foreach(int i in quantities){
            if(right<i){
                right = i;
            }
        }
        int res=0;
        while(left<=right){
            int mid = left+(right-left)/2;
            if(checker(n,quantities,mid)){
                res = mid;
                right = mid - 1;
            }else{
                left = mid + 1;
            }
        }
        return res;
    }
}